sidechain/contracts/group/Group.sol

```diff
@@ -309,31 +448,210 @@ contract Group is GroupI, UsingExternalStorage
         pure
         returns (bytes32)
     {
-        return keccak256(abi.encodePacked(_groupId, _role, _secretHash, _nonce));
+        return keccak256(
+            abi.encodePacked(
+                _groupId, 
+                "storeInvitation", // for sig uniqueness
+                _role, 
+                _secretHash, 
+                _nonce
+            )
+        );
     }
 
-    function inviteToCommunity(
+    // store invitation 
+    function storeInvitation(
```

Comment:
working on right now

Reply:
i still don't like how roles are instantiated in the constructor so i'm going to build a better solution for that now (i just never got around to it before), since you've got invite/storeInvitation() working, i'm going to turn it into a mapping so that the require() check on roles is easier